src/main/java/org/opentripplanner/routing/edgetype/StreetEdge.java

```diff
@@ -836,7 +840,7 @@ public void removeTurnRestriction(TurnRestriction turnRestriction) {
     synchronized (this) {
       if (turnRestrictions.contains(turnRestriction)) {
         if (turnRestrictions.size() == 1) {
-          turnRestrictions = List.of();
+          turnRestrictions = Collections.emptyList();
         } else {
           // in order to guarantee fast access without extra allocations
           // we make the turn restrictions unmodifiable after a copy-on-write modification
```

Comment:
Is it worth looking at if `List.copyOf(withRemoved)` could also use `Collection.unmodifiableList`? Or is there only a speed advantage for empy lists?

Reply:
So `Collections.emptyList()` have a static single iterator while does not `List.of()` - if this improves the performance then it needs a comment.